src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -1020,9 +1097,14 @@ void Instance::OnFinished(Status status, CharSpan debugText, WiFiScanResponseIte
         return;
     }
 
-    mLastNetworkingStatusValue.SetNonNull(status);
-    mLastConnectErrorValue.SetNull();
-    mLastNetworkIDLen = 0;
+    // If drivers are failing to respond NetworkNotFound on empty results, force it for them.
+    bool resultsMissing = !networks || (networks && (networks->Count() == 0));
```

Comment:
boolean operators should shortcut. `(networks == nullptr) || (networks->Count() == 0)` should be safe.

Reply:
Will do